.github/workflows/e2e-test-provider-example.yml

```
@@ -259,11 +297,137 @@ jobs:
             terraform apply -target module.azure_iam -auto-approve
             terraform apply -target module.azure_infrastructure -auto-approve
             ../build/constellation maa-patch "$(terraform output -raw maa_url)"
-            TF_LOG=INFO terraform apply -target constellation_cluster.azure_example -auto-approve
+            terraform apply -target constellation_cluster.azure_example -auto-approve
           else
-            TF_LOG=INFO terraform apply -auto-approve
+            terraform apply -auto-approve
+          fi
+
+      - name: Cleanup Terraform Cluster on failure
+        # cleanup here already on failure, because the subsequent TF overrides might make the TF config invalid and thus the destroy would fail later
+        # outcome is part of the steps context (https://docs.github.com/en/actions/learn-github-actions/contexts#steps-context)
+        if: failure() && steps.apply_terraform.outcome != 'skipped'
+        working-directory: ${{ github.workspace }}/cluster
+        shell: bash
+        run: |
+          terraform init
+          terraform destroy -auto-approve
+
+      - name: Add Provider to local Terraform registry # needed if release version was used before
+        if: inputs.providerVersion != ''
+        working-directory: ${{ github.workspace }}/build
+        shell: bash
+        run: |
+          bazel run //:devbuild --cli_edition=enterprise
+
+      - name: Update cluster configuration # for duplicate variable declaration, the last one is used
+        working-directory: ${{ github.workspace }}/cluster
+        shell: bash
+        run: |
+          if [[ "${{ inputs.toImage }}" != "" ]]; then
+          cat >> _override.tf <<EOF
+          locals {
+            image_version = "${{ inputs.toImage && inputs.toImage || steps.find-latest-image.outputs.image }}"
+          }
+          EOF
+          fi
```

I think this breaks the "upgrade to latest main if not set" behavior
The conditional should be removed, since we either have `inputs.toImage` as non empty, or `steps.find-latest-image.outputs.image`
(Need to double check the template, though I think it should be correct to assume it picks the second, if `inputs.toImage` is an empty string)
```suggestion
          cat >> _override.tf <<EOF
          locals {
            image_version = "${{ inputs.toImage || steps.find-latest-image.outputs.image }}"
          }
          EOF
```

---

good catch